scripts/taxonomy/GetTaxonomyTable.pl

```diff
@@ -0,0 +1,115 @@
+# Output directory where the taxonomy table will be saved
+my $out_dir = Cwd::abs_path();
+# Handle input arguments
+GetOptions("help"          => \$help,
+           "compara_url=s" => \$compara_url,
+           "outdir=s"      => \$out_dir
+          ) or die("Error in command line arguments\n");
+die "Error in command line arguments -compara_url " .
+    "mysql://user:pass\@host:port/db_name [-outdir /your/directory/]"
+    if (!$compara_url);
+
+# Create new Compara database adaptor and get the GenomeDB adaptor
+my $compara_dba = Bio::EnsEMBL::Compara::DBSQL::DBAdaptor->new(
+    -url => $compara_url, -species => 'Multi');
+my $genome_db_adaptor = $compara_dba->get_GenomeDBAdaptor();
+# Get all the Genome DBs
+my $list_ref_of_gdbs = $genome_db_adaptor->fetch_all();
+# Declare the taxonomic ranks of interest (sorted)
+my @target_ranks = qw(class superorder order suborder family subfamily species);
+# Create the taxonomy template: if a rank is missing, assign "N/A"
+my %rank_template = map {$_ => "N/A"} @target_ranks;
+# Default output file name
+my $out_file = "taxonomy_table.tsv";
+# Recursive function to flatten the taxonomy hash into an array with one string
+# per species' taxonomy (ranks separated by "\t"), and each rank sorted in
+# alphabetical order
+sub flatten {
+    my ($in_hash, $out_array, $already_flatted) = @_;
+    # The first call will have only two arguments
+    $already_flatted = "" unless(defined $already_flatted);
+    for my $key (sort keys %$in_hash) {
+        my $value = $in_hash->{$key};
+        if (ref $value eq 'HASH') {
+            flatten($value, $out_array, $already_flatted . $key . "\t");
+        } else {
+            push(@{$out_array}, $already_flatted . $key);
+        }
+    }
+}
+# Store all the taxonomic information in a mutidimensional hash, following the
+# same order as in @target_ranks
+my $taxonomy_db = {};
+foreach my $ref (@{$list_ref_of_gdbs}) {
+    my $node;
+    # There are some $ref that may not have 'taxon_id', so we need to handle the
+    # exception, report the troubling $ref and move to the next one
+    eval {
+        $node = $ref->taxon();
+    } or do {
+        warn "Undefined taxon_id for '", $ref->name(), "'.\n";
+        next;
+    };
+    # Copy taxonomy template and fill it in
+    my %taxonomy = %rank_template;
+    while ($node->name() ne 'root') {
+        my $rank = $node->rank();
+        if (defined $rank && exists $taxonomy{$rank}) {
+            $taxonomy{$rank} = $node->name();
+        }
+        $node = $node->parent();
+    }
+    # Replace species name by GenomeDB name
+    if (grep(/^species$/, @target_ranks)) {
+        $taxonomy{'species'} = $ref->name();
+    }
+    # Push the template into the taxonomy hash
+    my $level = $taxonomy_db;
+    foreach my $rank (@target_ranks) {
+        my $classification = $taxonomy{$rank};
+        if (! exists($level->{$classification})) {
+            # Each rank has a hash as value except "species" (this has a 1)
```

Comment:
By adding the genome_db name as an extra column at the end, I think you wouldn't need this special treatment here. Always set the value to `{}` and after the loop you may do ```$level->{$genome_db->name} = 1```